src/openforms/formio/rendering/default.py

```diff
@@ -69,6 +76,23 @@ def apply_to_labels(self, f: Callable[[str], str]) -> None:
         for choice in self.component["values"]:
             choice["label"] = f(choice["label"])
 
+    @property
+    def value_list(self):
+        if not self.component["type"] == "selectboxes":
+            return super().display_value
+
+        values = []
+        for value, isSelected in super().value.items():
+            if isSelected:
+                value_to_add = value
+                # Search the label for the value
+                for choice in self.component["values"]:
+                    if choice["value"] == value:
+                        value_to_add = choice["label"]
+
+                values.append(value_to_add)
+        return values
```

Comment:
This doesn't seem the right approach, `display_value` goes through `openforms.formio.service.format_value` which again uses the component registry and ultimately calls `openforms.formio.registry.BasePlugin.formatter` - see for example `src/openforms/formio/components/vanilla.py`.

If unspecified, the default formatter is used, and I think all of them are based on `FormatterBase`, which has access to the component and whether HTML is enabled or not. So, IMO the solution lies there to output `<ul>` HTML if HTML is enabled, and stick to the semi colon for plain text rendering.

Patching the code there should remove the need for the template changes and this new `value_list` concept.

FYI, the `value` is supposed to be the properly typed value, and whether it's a list or not is already encapsulated in there. The presentation layer `display_value` is its counterpart, which basically gives a "string representation of the value", optionally HTML capable if the context demands it.

Reply:
Check, yeah I thought it would be best to keep the html tags inside the html file. This whole `value_list` concept was a way to supplement that. But you're right, it just creates confusion and unexpected behavior/outputs...

I've changed it now so that the html is added in the formatters. This is a much nicer solution, and gives a lot more control. Most of the changes are in the `join_formatted_values` of `formio/formatters/base.py`. Here is decided which values should be returned as a html list. I've added it here, instead of in the individual formatters, because the formatters treat (in most cases) all values of a component individually (so to add the `<ul>` tags, we still would have to use the `join_formatted_values`).
I do have my doubts with this approach for the selectboxes component. The formatter for the selectboxes receives the entire values dict, and transforms it to a `;` separated string. Whereas the other components use the `join_formatted_values` to join the values together. To make this work, the selectboxes formatter now returns a string of `li` elements (when `as_html=True`)

Reply:
Sounds like what I initially assessed, nice to see that confirmed. I will take a closer look at the selectboxes later - but they have an odd data structure so I wouldn't be surprised if special treatment is still needed.

Reply:
Yeah i tried to figure out why the selectboxes formatter receives the entire dict, instead of 1 value from the dict, but i haven't found it out yet. Maybe we can change that, to keep all the formatters somewhat simular. 

But for now this works fine..
(Maybe the selectboxes formatter should also add the `ul` tags, to keep it simple/clear.. 🤔)